src/libraries/System.IO.Compression/src/System/IO/Compression/ZipArchiveEntry.cs

```diff
@@ -938,19 +988,40 @@ private bool WriteLocalFileHeader(bool isEmptyFile)
                 extraFieldLength = (ushort)bigExtraFieldLength;
             }
 
+            ZipLocalFileHeader.SignatureConstantBytes.CopyTo(lfStaticHeader);
+            currOffset += ZipLocalFileHeader.SignatureConstantBytes.Length;
+
+            BinaryPrimitives.WriteUInt16LittleEndian(lfStaticHeader.Slice(currOffset), (ushort)_versionToExtract);
+            currOffset += sizeof(ushort);
+
+            BinaryPrimitives.WriteUInt16LittleEndian(lfStaticHeader.Slice(currOffset), (ushort)_generalPurposeBitFlag);
+            currOffset += sizeof(ushort);
+
+            BinaryPrimitives.WriteUInt16LittleEndian(lfStaticHeader.Slice(currOffset), (ushort)CompressionMethod);
+            currOffset += sizeof(ushort);
+
+            BinaryPrimitives.WriteUInt32LittleEndian(lfStaticHeader.Slice(currOffset), ZipHelper.DateTimeToDosTime(_lastModified.DateTime));
+            currOffset += sizeof(uint);
+
+            BinaryPrimitives.WriteUInt32LittleEndian(lfStaticHeader.Slice(currOffset), _crc32);
+            currOffset += sizeof(uint);
+
+            BinaryPrimitives.WriteUInt32LittleEndian(lfStaticHeader.Slice(currOffset), compressedSizeTruncated);
+            currOffset += sizeof(uint);
+
+            BinaryPrimitives.WriteUInt32LittleEndian(lfStaticHeader.Slice(currOffset), uncompressedSizeTruncated);
+            currOffset += sizeof(uint);
+
+            BinaryPrimitives.WriteUInt16LittleEndian(lfStaticHeader.Slice(currOffset), (ushort)_storedEntryNameBytes.Length);
+            currOffset += sizeof(ushort);
+
+            BinaryPrimitives.WriteUInt16LittleEndian(lfStaticHeader.Slice(currOffset), extraFieldLength);
+            currOffset += sizeof(ushort);
```

Comment:
Here and elsewhere where a read or a write is done, please consider my suggestion of using constant field lengths and positions where they apply.

Obviously there are cases where the data length and position is variable, so that's where it's ok to do an operation.